### DISCUSSION TITLE
Prisma + supabase : handle references to auth.users

### DISCUSSION THREAD

**Discussion by NicolasChusseau** [prisma/prisma#19188] [Q&A] [answered]
Supabase has an auth.users table that lists all users with lots of info about them. 
I want to have a table (let's say profiles) that has its id as a foreign key to the id of the auth.users table.

Something like that : 
profiles (id, email, name)
- id is the primary key
- id is a foreign key that points to id of auth.users

auth.users is a table that cannot be modified in its structure.
The id that we want to refer is the primary key of the table.

For now, I have this error when I want to do a migration : 
![image](https://github.com/prisma/prisma/assets/122353423/0eddd8ac-bb8b-40d4-8ae6-900b1eb12d08)

**Comment by nurul3101** [comment, marked as answer]
Hey @NicolasChusseau 👋 

Have you enabled the [Multi Schema](https://www.prisma.io/docs/guides/other/multi-schema) Preview feature?
You would need to pass the auth schema in the datasource like this:

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["multiSchema"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
  schemas  = ["auth", "public"]
}
```

Can you try this and check?

**Reply by NicolasChusseau** [reply]
I tried with this schema.prisma : 
```prisma
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
  schemas  = ["auth", "public"]
}

generator client {
  provider      = "prisma-client-js"
  binaryTargets = "native"
  previewFeatures = ["multiSchema"]
}

model profiles {
  id         String   @id @default(uuid())
  updatedat  DateTime?
  firstName  String?
  lastName   String?
  email      String @unique

  @@schema("public")
}
```

But I have now a new error : 
![image](https://github.com/prisma/prisma/assets/122353423/62cff998-eea1-4db0-8a0f-c6e5fdd365ca)

**Reply by nurul3101** [reply]
It looks like you are running into this issue:
- https://github.com/prisma/prisma/issues/17734

Can you have a look at the issue and check some of the helpful comments there?

**Reply by NicolasChusseau** [reply]
Thanks <3,
I finally added storage to the array schemas and it works.
So now I have this : 
`schemas  = ["auth", "public", "storage"]`
But do I need to add something to my `.prisma` file to make the reference to the `auth.users` table explicit or is having it in supabase enough?

**Reply by nurul3101** [reply]
I am glad to hear that it's working. I don't think you need to add anything to the schema file now. Do you still get any error?

**Reply by NicolasChusseau** [reply]
Finally, I changed my mind as it was causing problems as the auth and storage tables were being deleted from supabase. I preferred to add a trigger that when a new person creates an account, it adds it to my table. So I no longer have a foreign key to auth.users. 
Even if I finally changed my mind, thank you very much for your help <3.

If anyone is interested, I'll show you the final code.
My .prisma file (I change the name from profiles to user) : 
```prisma
model user {
  id                  String       @id @db.Uuid @default(uuid())
  name                String?
  email               String    @unique
}
```

In supabase sql editor, run the following code after doing a migration : 
```sql
create function public.handle_new_user()
returns trigger as $$
begin
  insert into public."user" (id)
  values (new.id);
  return new;
end;
$$ language plpgsql security definer;
create trigger on_auth_user_created
  after insert on auth.users
  for each row execute procedure public.handle_new_user();
```

**Reply by nurul3101** [reply]
Thank you so much for sharing the final code! 🙌

**Reply by unhyif** [reply]
Hi @NicolasChusseau , so are you not synchronizing `auth.users` table with your prisma schema file? Thank you for your comments here!
